backend/adapter_processor/views.py

```diff
@@ -247,20 +247,33 @@ def partial_update(
             # if removed user use this adapter as default
             # Remove the same from his default
             for user_id in removed_users:
-                user_default_adapter = UserDefaultAdapter.objects.get(
-                    user_id=user_id
-                )
-
-                if user_default_adapter.default_llm_adapter == adapter:
-                    user_default_adapter.default_llm_adapter = None
-                elif user_default_adapter.default_embedding_adapter == adapter:
-                    user_default_adapter.default_embedding_adapter = None
-                elif user_default_adapter.default_vector_db_adapter == adapter:
-                    user_default_adapter.default_vector_db_adapter = None
-                elif user_default_adapter.default_x2text_adapter == adapter:
-                    user_default_adapter.default_x2text_adapter = None
-
-                user_default_adapter.save()
+                try:
+                    user_default_adapter = UserDefaultAdapter.objects.get(
+                        user_id=user_id
+                    )
+
+                    if user_default_adapter.default_llm_adapter == adapter:
+                        user_default_adapter.default_llm_adapter = None
+                    elif (
+                        user_default_adapter.default_embedding_adapter
+                        == adapter
+                    ):
+                        user_default_adapter.default_embedding_adapter = None
+                    elif (
+                        user_default_adapter.default_vector_db_adapter
+                        == adapter
+                    ):
+                        user_default_adapter.default_vector_db_adapter = None
+                    elif user_default_adapter.default_x2text_adapter == adapter:
+                        user_default_adapter.default_x2text_adapter = None
+
+                    user_default_adapter.save()
+                except UserDefaultAdapter.DoesNotExist:
```

Comment:
NIT: This exception would be caught only when calling `UserDefaultAdapter.objects.get()`. We could move this check over there

Reply:
I wanted to use `user_default_adapter`  in the below line,  else I would be required to initialize it outside the try block with none. Also additionally will require to add a null check before accessing the elements inside the variable.